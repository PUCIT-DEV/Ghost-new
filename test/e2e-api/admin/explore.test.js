const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyEtag, anyISODate, anyISODateTime, stringMatching} = matchers;

describe('Explore API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('members');
        await agent.loginAsOwner();
    });

    describe('Read', function () {
        it('Can request Explore data', async function () {
            await agent
                .get('explore/')
                .expectStatus(200)
                .matchBodySnapshot({
                    explore: {
                        mostRecentlyPublishedAt: anyISODateTime,
                        mrrStats: {
                            data: [{
                                date: anyISODate
                            },
                            {
                                date: anyISODate
                            }]
                        },
                        version: stringMatching(/\d+\.\d+\.\d+/)
                    }
                })
                .matchHeaderSnapshot({
                    etag: anyEtag
                });
        });
    });
});
