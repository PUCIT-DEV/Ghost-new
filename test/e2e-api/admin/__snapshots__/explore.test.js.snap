// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore API Read Can request Explore data 1: [body] 1`] = `
Object {
  "explore": Object {
    "mostRecentlyPublishedAt": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
    "mrrStats": Object {
      "data": Array [
        Object {
          "currency": "usd",
          "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
          "mrr": 0,
        },
        Object {
          "currency": "usd",
          "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
          "mrr": 1000,
        },
      ],
      "meta": Object {
        "totals": Array [
          Object {
            "currency": "usd",
            "mrr": 1000,
          },
        ],
      },
    },
    "ownerEmail": "jbloggs@example.com",
    "site": Object {
      "description": "Thoughts, stories and ideas",
      "icon": null,
      "title": "Ghost",
      "url": "http://127.0.0.1:2369/",
    },
    "stripe": Object {
      "configured": true,
      "livemode": false,
    },
    "totalMembers": 8,
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Explore API Read Can request Explore data 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "463",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
